building/build-debs/homeworld-admin-tools/src/version.py

```diff
@@ -0,0 +1,10 @@
+import os
+from resources import get_resource
+
+
+def display_version():
+    deb_version = get_resource("DEB_VERSION").decode("utf-8").rstrip()
+    print("Debian package version:", deb_version)
+
+    git_version = get_resource("GIT_VERSION").decode("utf-8").rstrip()
+    print("Build commit hash:", git_version)
```

Comment:
We generally don't bother with specifying the codec when we call `.decode()`. Do we have a particular new reason to do that?

Also, does this correctly handle the case where we build directly with inner_build.sh and resources/GIT_VERSION is missing?

Reply:
Regarding your first point: I did not realize decode defaulted to utf-8. Will change the code to not specify codec.

Reply:
Regarding your second point: the code crashes when building only with inner-build.sh. What should be the behavior when just building with inner-build? Should it behave the same as outer-build?

Reply:
Well, it should *definitely* not crash! Especially not delayed until runtime like that. A long-distance crash caused by building a different way is an unfortunate bug.

We also do call ./inner-build.sh directly during testing, so we should definitely support this condition. We should probably populate the resource in all cases.

With regards to the debian version, it probably makes more sense to only populate it when using ./build_package.sh, and then say "this was not built as part of a debian package" if ./inner_build.sh was used, either by detecting the lack of the resource or by sticking that string into the file.

It's probably worth, in any case, making sure that there's a clear error message if either of these is missing. It might be clear enough to just say "GIT_VERSION cannot be found" or something, which might already be the error message, but if the error message is more confusing than that, it might be worth adding extra code for a better message.